src/Microsoft.ML.PipelineInference/PipelinePattern.cs

```diff
@@ -183,34 +194,55 @@ public Experiment CreateTrainTestExperiment(IDataView trainData, IDataView testD
         /// <summary>
         /// Runs a train-test experiment on the current pipeline, through entrypoints.
         /// </summary>
-        public double RunTrainTestExperiment(IDataView trainData, IDataView testData, AutoInference.SupportedMetric metric, MacroUtils.TrainerKinds trainerKind)
+        public Tuple<double, double> RunTrainTestExperiment(IDataView trainData, IDataView testData, AutoInference.SupportedMetric metric, MacroUtils.TrainerKinds trainerKind)
```

Comment:
>Tuple<double, double> [](start = 15, length = 21)

Hi George, now that this is returning multiple values, do you think it might be nice to restructure this a little so that it is using `out` parameters? My reasoning is this will make the code more clear.

I'll tell you why I think this: I started in `AutoInference.cs`, where I saw this tuple being created. I wondered what was in it. But the usage there didn't help me, specifically:

```csharp
double d = result.Item1;
double d2 = result.Item2;
```

I eventually made my way here, but there was no documentation. I read through and eventually found it 

If there were two `out` parameters, one named `metricValue` and one named `trainingMetricValue`, then the method would be (to some extent) more self-documenting.

For this reason and some others we've tended to avoid the use of `Tuple`. (There happens to be a [lovely new concept of tuples in C# 7](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#tuples) that solves most of what we think of as tuples problems, but using that is perhaps a bit more involved, requiring subscription to nugets etc. we may not want to commit to right now.)

Reply:
Good idea. I'll make the suggested change.